src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1568,6 +1576,21 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 ->setArguments(array($transport['dsn'], $transport['options']))
                 ->addTag('messenger.receiver', array('alias' => $name))
             ;
+
+            if (0 === strpos($transport['dsn'], 'symfony://kernel.terminate')) {
```

Comment:
This will allow extra characters in the dsn, is that the expected behaviour?

Reply:
Oh, I see we can pass the bus as an option and in the query string

Reply:
Yes indeed. Trying to answer @sroze review https://github.com/symfony/symfony/pull/28746#discussion_r229441542 asking for both DSN query params & transport options support.